neat_core.c

```diff
@@ -804,6 +1035,210 @@ neat_open(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port)
+            ((unsigned int)seconds);
+#else
+            ((unsigned int)seconds) * 1000; // seconds -> ms
+#endif
+
+        int rc = setsockopt(flow->fd, IPPROTO_TCP,
+#if defined(__FreeBSD__) || defined(__NetBSD__)
+	    // KAH: turns out this was implemented on a feature
+	    // branch, and sadly not merged to mainline
+                TCP_SNDUTO_TIMEOUT,
+#else
+                TCP_USER_TIMEOUT,
+#endif
+                &new_value,
+                sizeof(new_value));
+
+        if (rc < 0) {
+            neat_log(NEAT_LOG_ERROR,
+                    "Unable to change timeout for TCP socket: "
+                    "Call to setsockopt failed with errno=%d", errno);
+            return NEAT_ERROR_IO;
+        }
+
+        return NEAT_ERROR_OK;
+    } else if (flow->sockProtocol == IPPROTO_SCTP) {
+#ifdef USRSCTP_SUPPORT
+        unsigned int new_value = ((unsigned int)seconds) * 1000 / 4; // seconds -> ms
+        usrsctp_sysctl_set_sctp_heartbeat_interval_default((uint32_t)new_value);
+        usrsctp_sysctl_set_sctp_assoc_rtx_max_default(4);
+#else
+        struct sctp_paddrparams params;
+        unsigned int optsize = sizeof(params);
+        int rc = getsockopt(flow->fd, IPPROTO_SCTP, SCTP_PEER_ADDR_PARAMS, &params, &optsize);
+        if (rc < 0) {
+            neat_log(NEAT_LOG_ERROR,
+                    "Unable to change timeout for TCP socket: "
+                    "Call to getsockopt failed with errno=%d", errno);
+            return NEAT_ERROR_IO;
+        }
+
+        params.spp_hbinterval = (unsigned int)seconds * 1000 / 4;
+        params.spp_flags |= SPP_HB_ENABLE;
+        params.spp_pathmaxrxt = 4;
+
+        rc = setsockopt(flow->fd, IPPROTO_SCTP, SCTP_PEER_ADDR_PARAMS, &params, sizeof(params));
+        if (rc < 0) {
+            neat_log(NEAT_LOG_ERROR,
+                    "Unable to change timeout for TCP socket: "
+                    "Call to setsockopt failed with errno=%d", errno);
+            return NEAT_ERROR_IO;
+        }
+#endif
+        return NEAT_ERROR_OK;
+    }
+#endif // !(defined(__FreeBSD__) || defined(__NetBSD__) || defined (__APPLE__))
+
+
+    return NEAT_ERROR_UNABLE;
+}
+
+static void
+set_primary_dest_resolve_cb(struct neat_resolver *resolver, struct neat_resolver_results *results, uint8_t code)
+{
+    neat_flow *flow = (neat_flow *)resolver->userData1;
+    struct neat_ctx *ctx = flow->ctx;
+    neat_log(NEAT_LOG_DEBUG, "%s", __func__);
+
+    if (code != NEAT_RESOLVER_OK) {
+        io_error(ctx, flow, code);
+        return;
+    }
+
+    assert(results->lh_first);
+
+#ifdef USRSCTP_SUPPORT
+    struct sctp_setprim addr;
+    addr.ssp_addr = results->lh_first->dst_addr;
+
+    if (usrsctp_setsockopt(flow->sock, IPPROTO_SCTP, SCTP_PRIMARY_ADDR, &addr, sizeof(addr)) < 0) {
+        neat_log(NEAT_LOG_DEBUG, "Call to usrsctp_setsockopt failed");
+        return;
+    }
+#elif defined(HAVE_NETINET_SCTP_H) && defined(__linux__)
+    struct sctp_prim addr;
+    struct sctp_assocparams assocparams;
```

Comment:
I personally find this to be an unnecessary restriction, and I don't see any technical reason for it unless we're going to support C89/ANSI C as well as C99, which is what the cmake file specifies.

I don't have any major objections, but it would be nice to have a well-defined coding style for the project, which I think is a point that's to be discussed in the plenary in Oslo if I'm not mistaken.